Compilation errors:

```
> Task :compileJava
/home/travis/build/ReactiveX/RxJava/src/main/java/io/reactivex/rxjava3/core/Flowable.java:19807: error: cannot infer type arguments for FlowableCollectWithCollectorSingle<>
        return RxJavaPlugins.onAssembly(new FlowableCollectWithCollectorSingle<>(this, collector));
                                        ^
  reason: inferred type does not conform to equality constraint(s)
    inferred: CAP#1
    equality constraints(s): CAP#1,T
  where T is a type-variable:
    T extends Object declared in class Flowable
  where CAP#1 is a fresh type-variable:
    CAP#1 extends Object super: T from capture of ? super T
/home/travis/build/ReactiveX/RxJava/src/main/java/io/reactivex/rxjava3/core/Observable.java:16647: error: cannot infer type arguments for ObservableCollectWithCollectorSingle<>
        return RxJavaPlugins.onAssembly(new ObservableCollectWithCollectorSingle<>(this, collector));
                                        ^
  reason: inferred type does not conform to equality constraint(s)
    inferred: CAP#1
    equality constraints(s): CAP#1,T
  where T is a type-variable:
    T extends Object declared in class Observable
  where CAP#1 is a fresh type-variable:
    CAP#1 extends Object super: T from capture of ? super T
2 errors
```

--- a/src/main/java/io/reactivex/rxjava3/core/Flowable.java
+++ b/src/main/java/io/reactivex/rxjava3/core/Flowable.java
@@ -19802,7 +19802,7 @@ public abstract class Flowable<@NonNull T> implements Publisher<T> {
     @BackpressureSupport(BackpressureKind.UNBOUNDED_IN)
     @SchedulerSupport(SchedulerSupport.NONE)
     @NonNull
-    public final <@NonNull R, A> Single<R> collect(@NonNull Collector<T, A, R> collector) {
+    public final <@NonNull R, A> Single<R> collect(@NonNull Collector<? super T, A, R> collector) {
         Objects.requireNonNull(collector, "collector is null");
         return RxJavaPlugins.onAssembly(new FlowableCollectWithCollectorSingle<>(this, collector));
     }

--- a/src/main/java/io/reactivex/rxjava3/core/Observable.java
+++ b/src/main/java/io/reactivex/rxjava3/core/Observable.java
@@ -16642,7 +16642,7 @@ public abstract class Observable<@NonNull T> implements ObservableSource<T> {
     @CheckReturnValue
     @SchedulerSupport(SchedulerSupport.NONE)
     @NonNull
-    public final <@NonNull R, A> Single<R> collect(@NonNull Collector<T, A, R> collector) {
+    public final <@NonNull R, A> Single<R> collect(@NonNull Collector<? super T, A, R> collector) {
         Objects.requireNonNull(collector, "collector is null");
         return RxJavaPlugins.onAssembly(new ObservableCollectWithCollectorSingle<>(this, collector));
     }

--- a/src/main/java/io/reactivex/rxjava3/internal/jdk8/FlowableCollectWithCollector.java
+++ b/src/main/java/io/reactivex/rxjava3/internal/jdk8/FlowableCollectWithCollector.java
@@ -36,9 +36,9 @@ public final class FlowableCollectWithCollector<T, A, R> extends Flowable<R> {
 
     final Flowable<T> source;
 
-    final Collector<T, A, R> collector;
+    final Collector<? super T, A, R> collector;
 
-    public FlowableCollectWithCollector(Flowable<T> source, Collector<T, A, R> collector) {
+    public FlowableCollectWithCollector(Flowable<T> source, Collector<? super T, A, R> collector) {
         this.source = source;
         this.collector = collector;
     }
@@ -46,7 +46,7 @@ public final class FlowableCollectWithCollector<T, A, R> extends Flowable<R> {
     @Override
     protected void subscribeActual(@NonNull Subscriber<? super R> s) {
         A container;
-        BiConsumer<A, T> accumulator;
+        BiConsumer<A, ? super T> accumulator;
         Function<A, R> finisher;
 
         try {

--- a/src/main/java/io/reactivex/rxjava3/internal/jdk8/FlowableCollectWithCollectorSingle.java
+++ b/src/main/java/io/reactivex/rxjava3/internal/jdk8/FlowableCollectWithCollectorSingle.java
@@ -39,9 +39,9 @@ public final class FlowableCollectWithCollectorSingle<T, A, R> extends Single<R>
 
     final Flowable<T> source;
 
-    final Collector<T, A, R> collector;
+    final Collector<? super T, A, R> collector;
 
-    public FlowableCollectWithCollectorSingle(Flowable<T> source, Collector<T, A, R> collector) {
+    public FlowableCollectWithCollectorSingle(Flowable<T> source, Collector<? super T, A, R> collector) {
         this.source = source;
         this.collector = collector;
     }
@@ -54,7 +54,7 @@ public final class FlowableCollectWithCollectorSingle<T, A, R> extends Single<R>
     @Override
     protected void subscribeActual(@NonNull SingleObserver<? super R> observer) {
         A container;
-        BiConsumer<A, T> accumulator;
+        BiConsumer<A, ? super T> accumulator;
         Function<A, R> finisher;
 
         try {

--- a/src/main/java/io/reactivex/rxjava3/internal/jdk8/ObservableCollectWithCollector.java
+++ b/src/main/java/io/reactivex/rxjava3/internal/jdk8/ObservableCollectWithCollector.java
@@ -36,9 +36,9 @@ public final class ObservableCollectWithCollector<T, A, R> extends Observable<R>
 
     final Observable<T> source;
 
-    final Collector<T, A, R> collector;
+    final Collector<? super T, A, R> collector;
 
-    public ObservableCollectWithCollector(Observable<T> source, Collector<T, A, R> collector) {
+    public ObservableCollectWithCollector(Observable<T> source, Collector<? super T, A, R> collector) {
         this.source = source;
         this.collector = collector;
     }
@@ -46,7 +46,7 @@ public final class ObservableCollectWithCollector<T, A, R> extends Observable<R>
     @Override
     protected void subscribeActual(@NonNull Observer<? super R> observer) {
         A container;
-        BiConsumer<A, T> accumulator;
+        BiConsumer<A, ? super T> accumulator;
         Function<A, R> finisher;
 
         try {

--- a/src/main/java/io/reactivex/rxjava3/internal/jdk8/ObservableCollectWithCollectorSingle.java
+++ b/src/main/java/io/reactivex/rxjava3/internal/jdk8/ObservableCollectWithCollectorSingle.java
@@ -36,9 +36,9 @@ public final class ObservableCollectWithCollectorSingle<T, A, R> extends Single<
 
     final Observable<T> source;
 
-    final Collector<T, A, R> collector;
+    final Collector<? super T, A, R> collector;
 
-    public ObservableCollectWithCollectorSingle(Observable<T> source, Collector<T, A, R> collector) {
+    public ObservableCollectWithCollectorSingle(Observable<T> source, Collector<? super T, A, R> collector) {
         this.source = source;
         this.collector = collector;
     }
@@ -51,7 +51,7 @@ public final class ObservableCollectWithCollectorSingle<T, A, R> extends Single<
     @Override
     protected void subscribeActual(@NonNull SingleObserver<? super R> observer) {
         A container;
-        BiConsumer<A, T> accumulator;
+        BiConsumer<A, ? super T> accumulator;
         Function<A, R> finisher;
 
         try {

